### PR TITLE
computeBackendWeights: avoid rounding issues

### PR DESCRIPTION
**Pull request by aermakov-zalando**
The last active backend (with a non-zero traffic weight) can get a weight of less than 1.0 due to floating point precision issues. Fix by forcibly setting a weight of 1.0.

### CODE REVIEW

**Comment by mikkeloscar** [comment]
:+1:

**Comment by szuecs** [comment]
:-1:

File: dataclients/kubernetes/kube.go [hunk 1]
```diff
@@ -733,31 +733,41 @@ func catchAllRoutes(routes []*eskip.Route) bool {
...
 		if !ok {
-			sc = &sumCount{}
-			pathSumCount[path.Path] = sc
+			sc = &pathInfo{}
+			pathInfos[path.Path] = sc
 		}
 
 		if weight, ok := backendWeights[path.Backend.ServiceName]; ok {
 			sc.sum += weight
+			if weight > 0 {
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: why do you need this check?

File: dataclients/kubernetes/kube.go [hunk 2]
```diff
@@ -733,31 +733,41 @@ func catchAllRoutes(routes []*eskip.Route) bool {
 //
 // where for a weight of 1.0 no Traffic predicate will be generated.
 func computeBackendWeights(backendWeights map[string]float64, rule *rule) {
-	type sumCount struct {
-		sum   float64
-		count int
+	type pathInfo struct {
+		sum        float64
+		lastActive *backend
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: I think you don't need this lastActive to achieve the goal of having the last set to 1.0

File: dataclients/kubernetes/kube.go (hunk 1 above)
Comment (aermakov-zalando) [reply dataclients/kubernetes/kube.go]: To skip backends with weights explicitly set to 0.0.

File: dataclients/kubernetes/kube.go (hunk 2 above)
Comment (aermakov-zalando) [reply dataclients/kubernetes/kube.go]: Any suggestions, or at least a comment on why you don't like this approach?

**Comment by szuecs** [comment]
I am not convinced that this can happen right now and I think this could break other patterns.
Please provide examples or best test cases, which will show the problem that you found.

**Comment by aermakov-zalando** [comment]
I've added a test that was failing before the fix.

**Comment by szuecs** [comment]
Offline discussed with @aermakov-zalando, this will not break any ingress with predicate traffic and fixes a current float rounding issue.

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**